### PR TITLE
CSCFAIRMETA-434: [ADD] Drafts-Statistics-API-SQL-queries-should-be-mo…

### PR DESCRIPTION
**Pull request by katrite**
…dified-to-only-count-published-datasets

Add sql filters to rpc/statistics APIs so that drafts are not taken into account

### CODE REVIEW

File: src/metax_api/tests/api/rpc/base/views/statistic_rpc.py [hunk 1]
```diff
@@ -678,3 +693,89 @@ def test_all_datasets_cumulative_latest_legacy(self):
...
+    Tests that drafts are not taken into account when calculating statistics
+    """
+    def test_count_datasets_api_for_drafts(self):
+        """
+        Tests that rpc/statistics/count_datasets returns only count of published datasets
+        """
+        response_1 = self.client.get('/rpc/statistics/count_datasets').data
+
+        self._set_dataset_as_draft(1)
+        self.assertEqual(CatalogRecord.objects.get(pk=1).state, 'draft', CatalogRecord.objects.get(pk=1).id)
```
Comment (hannu40k) [inline src/metax_api/tests/api/rpc/base/views/statistic_rpc.py:709]: The last parameter does not make much sense, its just outputting id which is 1 always?

File: src/metax_api/tests/api/rpc/base/views/statistic_rpc.py [hunk 2]
```diff
@@ -678,3 +693,89 @@ def test_all_datasets_cumulative_latest_legacy(self):
...
+        'Drafts should not be returned in count_datasets api')
+
+    def test_all_datasets_cumulative_for_drafts(self):
+        """
+        Tests that /rpc/statistics/all_datasets_cumulative returns only published datasets
+        """
+        url = '/rpc/statistics/all_datasets_cumulative?from_date=2019-06&to_date=2019-06'
+
+        self._set_dataset_creation_date(1, '2019-06-15')
+        response_1 = self.client.get(f'{url}').data
```
Comment (hannu40k) [inline src/metax_api/tests/api/rpc/base/views/statistic_rpc.py:722]: ``f'{url}'`` is a silly way to use the variable url, f-strings should be used if some variable value should be formatted inside a string, but if the variable is a string and there is no other content in the target string, then its better to use the variable directly, e.g. ``self.client.get(url).data``.

File: src/metax_api/tests/api/rpc/base/views/statistic_rpc.py [hunk 3]
```diff
@@ -678,3 +693,89 @@ def test_all_datasets_cumulative_latest_legacy(self):
...
+        Tests that rpc/statistics/count_datasets returns only count of published datasets
+        """
+        response_1 = self.client.get('/rpc/statistics/count_datasets').data
+
+        self._set_dataset_as_draft(1)
+        self.assertEqual(CatalogRecord.objects.get(pk=1).state, 'draft', CatalogRecord.objects.get(pk=1).id)
+
+        response_2 = self.client.get('/rpc/statistics/count_datasets').data
+        self.assertNotEqual(response_1['count'], response_2['count'],
+        'Drafts should not be returned in count_datasets api')
```
Comment (hannu40k) [inline src/metax_api/tests/api/rpc/base/views/statistic_rpc.py:713]: If a function call needs to be continued on a second line, indent the second line once.
```        
self.assertNotEqual(response_1['count'], response_2['count'],
    'Drafts should not be returned in count_datasets api')
```

File: src/metax_api/tests/api/rpc/base/views/statistic_rpc.py (hunk 1 above)
Comment (katrite) [reply src/metax_api/tests/api/rpc/base/views/statistic_rpc.py:709]: True, changed it now to a text instead..

File: src/metax_api/tests/api/rpc/base/views/statistic_rpc.py (hunk 2 above)
Comment (katrite) [reply src/metax_api/tests/api/rpc/base/views/statistic_rpc.py:722]: ok thanks for noting

File: src/metax_api/tests/api/rpc/base/views/statistic_rpc.py (hunk 3 above)
Comment (katrite) [reply src/metax_api/tests/api/rpc/base/views/statistic_rpc.py:713]: Ok. Second line is needed only because flake doesn't allow too long lines.

**Pull request merged**